Add Clinical Records

PSchmiedmayer opened the pull request:
# Add Clinical Records

## :recycle: Current situation & Problem
- As noted in #39 we currently do not support all possible HealthKit sample types.


## :gear: Release Notes 
- This PR takes a step closer, adding clinical record types.


## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

lukaskollmer reviewed (changes requested):
very nice addition. i can look into the other HKSample subclasses over the next couple days. 

two thoughts:
1. can we somehow test this? iirc there are some sample types which can't be created by us, be it manually via Health.app or programmatically via the APIs (eg `HKDocumentType` falls into this category, i believe ). does this also apply to clinical records?
2. i'm noticing that SampleTypes.swift is updated, but the PR doesn't include any changes to SampleTypes.swift.gyb, is that just missing? we should make any changes to the pre-defined sample type accessors through that; otherwise the next gyb rerun will override the properties added here and three clinical record types will get removed

PSchmiedmayer commented:
Thank you for the catches on the gyb file; fixed this 🚀

Regarding testing: Clinical records are challenging. You can not create them in an app yourself, they can only be created in the Health App (like ECGs). There is an option to add a mock institution in the health app in the simulator. But even the process to do that + the permission sheet are different than the typical flow.

I think it might make sense to invest time time in the testing for this in the future; I experimented a bit with the new Spezi HealthKit version for LLMonFHIR and added these sample types for now, looks really nice with all the new additions 🚀

lukaskollmer reviewed (approved):
looks good now with the gyb adjustments. i made some additional changes, to bring the documentation in sync with the new sample types

PSchmiedmayer commented:
Sounds great; thank you! I fixed one small documentation element and would merge it for now. We can continue the discussion and work based on #39 and #41 to eventually update SpeziHealthKit in SpeziFHIR and LLMonFHIR down the line 🚀
